Question: How much are nearby Hospitals and Ambulatory Surgical Centers (surgery centers) paid for the same procedure / in the same area?

In [1]:
import pandas as pd 
import numpy as np  
import matplotlib.pyplot as plt  
import seaborn as sns

In [2]:
%matplotlib inline

d1 = Medicare Provider Utilization and Payment Data: Physician and Other Supplier PUF CY2017

In [20]:
pd.read_csv(
        'c:/users/erinm/Documents/healthcare-bluebook-green-team/data/Medicare_Provider_Utilization_and_Payment_Data__Physician_and_Other_Supplier_PUF_CY2017.tsv', sep='\t').dtypes
#d1.shape

National Provider Identifier                                  int64
Last Name/Organization Name of the Provider                  object
First Name of the Provider                                   object
Middle Initial of the Provider                               object
Credentials of the Provider                                  object
Gender of the Provider                                       object
Entity Type of the Provider                                  object
Street Address 1 of the Provider                             object
Street Address 2 of the Provider                             object
City of the Provider                                         object
Zip Code of the Provider                                     object
State Code of the Provider                                   object
Country Code of the Provider                                 object
Provider Type                                                object
Medicare Participation Indicator                

Note: for now, I only read in National Provider Identifer (and not all the NPI subcolumns, like name, address, etc., since I figured those could be brought in later if we are interested.)

In [4]:
d1=pd.read_csv(
        'c:/users/erinm/Documents/healthcare-bluebook-green-team/data/Medicare_Provider_Utilization_and_Payment_Data__Physician_and_Other_Supplier_PUF_CY2017.tsv', 
        sep='\t',
        usecols=['National Provider Identifier', 'Entity Type of the Provider', 'Provider Type', 'Place of Service', 'HCPCS Code', 'HCPCS Description', 'Number of Distinct Medicare Beneficiary/Per Day Services', 'Average Medicare Allowed Amount']
)
d1.shape

(9847443, 8)

In [5]:
d1.head()

National Provider Identifier Entity Type of the Provider  \
0                    1003000126                           I   
1                    1003000126                           I   
2                    1003000126                           I   
3                    1003000126                           I   
4                    1003000126                           I   

       Provider Type Place of Service HCPCS Code  \
0  Internal Medicine                F      99217   
1  Internal Medicine                F      99218   
2  Internal Medicine                F      99219   
3  Internal Medicine                F      99220   
4  Internal Medicine                F      99221   

                                   HCPCS Description  \
0                Hospital observation care discharge   
1     Hospital observation care typically 30 minutes   
2     Hospital observation care typically 50 minutes   
3  Hospital observation care typically 70 minutes...   
4  Initial hospital inpatient care, typically 30 ...   

   Number of Distinct Medicare Beneficiary/Per Day Services  \
0                                                100          
1                                                 26          
2                                                 52          
3                                                 59          
4                                                 16          

   Average Medicare Allowed Amount  
0                        73.398800  
1                       100.080000  
2                       136.380000  
3                       190.363729  
4                       101.680000

d2 = Provider Outpatient Hospital Charge Data by APC, CY2017

In [28]:
d2 = pd.read_csv(
    'c:/users/erinm/Documents/healthcare-bluebook-green-team/data/Provider_Outpatient_Hospital_Charge_Data_by_APC__CY2017.tsv', 
    sep='\t',
    usecols=['Provider_ID','APC','APC_Desc', 'Beneficiaries', 'CAPC_Services','Average_Medicare_Allowed_Amount']
)
d2.shape

(61779, 6)

In [29]:
d2.head()

Provider_ID   APC                                           APC_Desc  \
0        10001  5072    Level 2 Excision/ Biopsy/ Incision and Drainage   
1        10001  5073    Level 3 Excision/ Biopsy/ Incision and Drainage   
2        10001  5091  Level 1 Breast/Lymphatic Surgery and Related P...   
3        10001  5092  Level 2 Breast/Lymphatic Surgery and Related P...   
4        10001  5112                 Level 2 Musculoskeletal Procedures   

   Beneficiaries  CAPC_Services  Average_Medicare_Allowed_Amount  
0          249.0            259                          1038.45  
1           52.0             53                          1792.60  
2           26.0             27                          2113.58  
3           23.0             23                          3737.14  
4           17.0             17                          1029.46

In [30]:
d2.columns

Index(['Provider_ID', 'APC', 'APC_Desc', 'Beneficiaries', 'CAPC_Services',
       'Average_Medicare_Allowed_Amount'],
      dtype='object')

d3 = APC to CPT/HCPCS crosswalk (Jan 2020 Addendum B)

In [31]:
d3 = pd.read_excel('c:/users/erinm/Documents/healthcare-bluebook-green-team/data/2020_january_web_addendum_b.12312019.xlsx',usecols=[0,1,3], skiprows=2)
d3.head()

HCPCS Code            Short Descriptor  APC 
0      00100       Anesth salivary gland   NaN
1      00102  Anesth repair of cleft lip   NaN
2      00103       Anesth blepharoplasty   NaN
3      00104         Anesth electroshock   NaN
4      00120          Anesth ear surgery   NaN

d4 = Zip to CBSA crosswalk (Q1 2020)

In [32]:
d4 = pd.read_excel('c:/users/erinm/documents/healthcare-bluebook-green-team/data/ZIP_CBSA_032020.xlsx')
d4.head()

ZIP   CBSA  RES_RATIO  BUS_RATIO  OTH_RATIO  TOT_RATIO
0  501  35620        0.0        1.0        0.0        1.0
1  601  38660        1.0        1.0        1.0        1.0
2  602  10380        1.0        1.0        1.0        1.0
3  603  10380        1.0        1.0        1.0        1.0
4  604  10380        1.0        1.0        1.0        1.0

Inner join provider payment data (d1) with APC to HCPCS crosswalk (d3) on HCPCS Code.

In [33]:
d1_d3 = pd.merge(d1, d3, on=('HCPCS Code'), how='inner')
d1_d3.shape

(9502517, 10)

In [34]:
d1_d3.head()

National Provider Identifier Entity Type of the Provider  \
0                    1003000126                           I   
1                    1003000407                           I   
2                    1003006586                           I   
3                    1003007824                           I   
4                    1003008590                           I   

       Provider Type Place of Service HCPCS Code  \
0  Internal Medicine                F      99217   
1    Family Practice                F      99217   
2  Internal Medicine                F      99217   
3  Internal Medicine                F      99217   
4  Internal Medicine                F      99217   

                     HCPCS Description  \
0  Hospital observation care discharge   
1  Hospital observation care discharge   
2  Hospital observation care discharge   
3  Hospital observation care discharge   
4  Hospital observation care discharge   

   Number of Distinct Medicare Beneficiary/Per Day Services  \
0                                                100          
1                                                 15          
2                                                 53          
3                                                 12          
4                                                 28          

   Average Medicare Allowed Amount            Short Descriptor  APC   
0                          73.3988  Observation care discharge   NaN  
1                          72.3000  Observation care discharge   NaN  
2                          82.0900  Observation care discharge   NaN  
3                          79.6300  Observation care discharge   NaN  
4                          75.7300  Observation care discharge   NaN

Rename columns.

In [35]:
d1_d3.columns

Index(['National Provider Identifier', 'Entity Type of the Provider',
       'Provider Type', 'Place of Service', 'HCPCS Code', 'HCPCS Description',
       'Number of Distinct Medicare Beneficiary/Per Day Services',
       'Average Medicare Allowed Amount', 'Short Descriptor', 'APC '],
      dtype='object')

In [36]:
d1_d3.columns=['NPI', 
              'Entity Type', 
              'Provider Type', 
              'Place of Service', 
              'HCPCS Code', 
              'HCPCS Desc', 
              'No. Distinct Medicare Beneficiaries Per Day Services', 
              'Avg Medicare Amt Allowed', 
              'APC Desc', 
              'APC']
d1_d3.columns

Index(['NPI', 'Entity Type', 'Provider Type', 'Place of Service', 'HCPCS Code',
       'HCPCS Desc', 'No. Distinct Medicare Beneficiaries Per Day Services',
       'Avg Medicare Amt Allowed', 'APC Desc', 'APC'],
      dtype='object')

Remove entries with nans in APC Code field.

In [37]:
d1_d3_nn=d1_d3[d1_d3['APC'].notnull()]
d1_d3_nn.shape

(4016376, 10)

***Note: Tried to do the following but got memory error, so am moving join to after filters imposed.  

Full join d1_d3_nn (provider payment data (d1) with APC to HCPCS crosswalk (d3) on HCPCS Code) with Outpatient Charge data (d2) on APC.

In [40]:
d1_d3_nn.dtypes

NPI                                                       int64
Entity Type                                              object
Provider Type                                            object
Place of Service                                         object
HCPCS Code                                               object
HCPCS Desc                                               object
No. Distinct Medicare Beneficiaries Per Day Services      int64
Avg Medicare Amt Allowed                                float64
APC Desc                                                 object
APC                                                     float64
dtype: object

From PowerPoint: 

"Physician/Other Supplier Data Filters - pulling ASCs
entity type = 'O'
Provider Type = Ambulatory Surgical Center
CPT = 43249
Hospital Outpatient
APC = 5302
Knowing what they get paid for an APC, we can tell what they get paid for a procedure.
Compare results within a CBSA (what is a)"

Create filters:
filter1 = entity type (I or O)
filter2 = HCPCS Code
filter3 = provider type

In [41]:
filter1 = d1_d3_nn['Entity Type'].isin(['O'])
filter2 = d1_d3_nn['HCPCS Code'].isin(['43249'])
filter3 = d1_d3_nn['Provider Type'].isin(['Ambulatory Surgical Center'])

F1_F2_F3 = d1_d3_nn[filter1 & filter2 & filter3]
F1_F2_F3.shape

(616, 10)

In [42]:
F1_F2_F3.head()

NPI Entity Type               Provider Type Place of Service  \
7529288  1003015843           O  Ambulatory Surgical Center                F   
7529292  1003079187           O  Ambulatory Surgical Center                F   
7529293  1003220187           O  Ambulatory Surgical Center                F   
7529297  1003815242           O  Ambulatory Surgical Center                F   
7529307  1003887225           O  Ambulatory Surgical Center                F   

        HCPCS Code                                        HCPCS Desc  \
7529288      43249  Balloon dilation of esophagus using an endoscope   
7529292      43249  Balloon dilation of esophagus using an endoscope   
7529293      43249  Balloon dilation of esophagus using an endoscope   
7529297      43249  Balloon dilation of esophagus using an endoscope   
7529307      43249  Balloon dilation of esophagus using an endoscope   

         No. Distinct Medicare Beneficiaries Per Day Services  \
7529288                                                 20      
7529292                                                 17      
7529293                                                 94      
7529297                                                446      
7529307                                                 26      

         Avg Medicare Amt Allowed                   APC Desc     APC  
7529288                 814.53450  Esoph egd dilation <30 mm  5302.0  
7529292                 555.31000  Esoph egd dilation <30 mm  5302.0  
7529293                 596.51000  Esoph egd dilation <30 mm  5302.0  
7529297                 509.85778  Esoph egd dilation <30 mm  5302.0  
7529307                 576.83000  Esoph egd dilation <30 mm  5302.0

Confirming that Place of Service is uniform.

In [43]:
F1_F2_F3['Place of Service'].value_counts()

F    616
Name: Place of Service, dtype: int64

***Note: again tried to merge with d2 but am getting memory error.  Tried to limit with nrows (like Taylor suggested with a read_csv) but this does not work with the merge function.  Need to troubleshoot.

In [47]:
#F1_F2_F3_with_d2= pd.merge(d1_d3_nn, d2, on=('APC'), how='inner')
#F1_F2_F3_with_d2.shape

MemoryError: 

Assign new filters to d1_d3_nn.  Filters relate to our choice to look at hip replacements.  Know that APC code is [5112, 5113, 5114, 5115, 5116], which are various levels of muskoskeletal procedures.  Look at HCPCS within those APCs to find hip replacement code.

In [50]:
filter4 = d1_d3_nn['APC'].isin([5112, 5113, 5114, 5115, 5116])

F4 = d1_d3_nn[filter4]
F4.shape

(60478, 10)

In [53]:
F4['HCPCS Code'].value_counts()

27447    11213
27130     6780
63047     5039
29827     3593
26055     2678
         ...  
27080        1
25001        1
26455        1
26615        1
25300        1
Name: HCPCS Code, Length: 390, dtype: int64

In [54]:
F4['HCPCS Desc'].value_counts()

Repair of knee joint                                                              11807
Replacement of thigh bone and hip joint prosthesis                                 6780
Partial removal of middle spine bone with release of spinal cord and/or nerves     5047
Repair of shoulder rotator cuff using an endoscope                                 3593
Incision of tendon covering                                                        2678
                                                                                  ...  
Partial removal of leg bone                                                           1
Partial removal of hand bone                                                          1
Treatment of knee joint fractures with assistance of an endoscope                     1
Diagnostic examination of elbow joint using an endoscope                              1
Creation of web space between toes                                                    1
Name: HCPCS Desc, Length: 336, d

In [60]:
F5 = d1_d3_nn[d1_d3_nn['HCPCS Desc'].str.contains('hip')]

F5.shape

(63853, 10)

In [61]:
F5.head()

NPI Entity Type             Provider Type Place of Service  \
4902359  1003001785           I        Orthopedic Surgery                O   
4902360  1003002148           I      Diagnostic Radiology                F   
4902361  1003002809           I      Diagnostic Radiology                O   
4902362  1003006636           I      Diagnostic Radiology                F   
4902363  1003013129           I  Interventional Radiology                F   

        HCPCS Code                           HCPCS Desc  \
4902359      73502  X-ray of hip with pelvis, 2-3 views   
4902360      73502  X-ray of hip with pelvis, 2-3 views   
4902361      73502  X-ray of hip with pelvis, 2-3 views   
4902362      73502  X-ray of hip with pelvis, 2-3 views   
4902363      73502  X-ray of hip with pelvis, 2-3 views   

         No. Distinct Medicare Beneficiaries Per Day Services  \
4902359                                                 24      
4902360                                                 39      
4902361                                                 89      
4902362                                                 27      
4902363                                                 30      

         Avg Medicare Amt Allowed                      APC Desc     APC  
4902359                     26.91  X-ray exam hip uni 2-3 views  5521.0  
4902360                     11.19  X-ray exam hip uni 2-3 views  5521.0  
4902361                     47.39  X-ray exam hip uni 2-3 views  5521.0  
4902362                     11.39  X-ray exam hip uni 2-3 views  5521.0  
4902363                     12.04  X-ray exam hip uni 2-3 views  5521.0

In [69]:
F5['HCPCS Desc'].value_counts()

X-ray of hip with pelvis, 2-3 views                                                           39763
X-ray of hip with pelvis, 1 view                                                               6817
Replacement of thigh bone and hip joint prosthesis                                             6780
X-ray of both hips with pelvis, 2 views                                                        3797
X-ray of both hips with pelvis, 3-4 views                                                      3629
X-ray of both hips with pelvis, minimum of 5 views                                             1929
X-ray of hip with pelvis, minimum of 4 views                                                    713
Radiological supervision and interpretation X-ray of hip joint                                  250
Strapping of hip                                                                                 66
Removal or shaving of hip joint socket cartilage using an endoscope                              12


In [64]:
F5['APC'].value_counts()

5521.0    46580
5522.0    10068
5115.0     6780
5572.0      250
5733.0       66
5114.0       35
5113.0       35
5112.0       18
5111.0        9
5073.0        6
5072.0        4
5102.0        1
5054.0        1
Name: APC, dtype: int64

In [67]:
F5['APC Desc'].value_counts()

X-ray exam hip uni 2-3 views    39763
X-ray exam hip uni 1 view        6817
Total hip arthroplasty           6780
X-ray exam hips bi 2 views       3797
X-ray exam hips bi 3-4 views     3629
X-ray exam hips bi 5/> views     1929
X-ray exam hip uni 4/> views      713
Contrast x-ray of hip             250
Strapping of hip                   66
Incision of hip tendon             15
Hip arthr0 w/debridement           12
Hip arthro w/femoroplasty          11
Incision of hip tendons            10
Pelvis/hip joint surgery            8
Drainage of pelvis lesion           7
Remove hip bone les deep            6
Hip arthro w/labral repair          6
Manipulation of hip joint           6
Hip arthr0 w/synovectomy            5
Biopsy of soft tissues              4
Exc hip/pelv tum deep 5 cm/>        2
Transfer of spinal muscle           2
Hip arthro acetabuloplasty          2
Exploration of hip joint            2
Exc hip/pelv tum deep < 5 cm        2
Resect hip/pelv tum < 5 cm          1
Treat hip so

If we want hip replacements, I think our best bet is the following, which has 6780 occurrances in the dataset 1: HCPCS Code = 27130, APC Code = 5115, APC Desc on dataset 2 = Level 5 Musculoskeletal Procedure, APC Short Desc on crosswalk = Total Hip Arthroplasty.

In [72]:
filter6 = d1_d3_nn['APC Desc'].isin(['Total hip arthroplasty'])

F6 = d1_d3_nn[filter6]
F6.shape

(6780, 10)

In [73]:
F6.head()

NPI Entity Type        Provider Type Place of Service  \
6748341  1003006115           I  Physician Assistant                F   
6748342  1003017682           I   Orthopedic Surgery                F   
6748343  1003018086           I   Orthopedic Surgery                F   
6748344  1003029034           I   Orthopedic Surgery                F   
6748345  1003044033           I   Orthopedic Surgery                F   

        HCPCS Code                                         HCPCS Desc  \
6748341      27130  Replacement of thigh bone and hip joint prosth...   
6748342      27130  Replacement of thigh bone and hip joint prosth...   
6748343      27130  Replacement of thigh bone and hip joint prosth...   
6748344      27130  Replacement of thigh bone and hip joint prosth...   
6748345      27130  Replacement of thigh bone and hip joint prosth...   

         No. Distinct Medicare Beneficiaries Per Day Services  \
6748341                                                 14      
6748342                                                 22      
6748343                                                 15      
6748344                                                 35      
6748345                                                 14      

         Avg Medicare Amt Allowed                APC Desc     APC  
6748341                230.510000  Total hip arthroplasty  5115.0  
6748342               1376.500000  Total hip arthroplasty  5115.0  
6748343               1339.390000  Total hip arthroplasty  5115.0  
6748344               1260.513714  Total hip arthroplasty  5115.0  
6748345               1283.370000  Total hip arthroplasty  5115.0

Confirming that entity type is uniform.  Confirmed.  All are "I".

In [74]:
F6['Entity Type'].value_counts()

I    6780
Name: Entity Type, dtype: int64

In [ ]:
Confirming that Place of Service is uniform.  No.  8 occurrances are 'O'.  Need to remove those.

In [75]:
F6['Place of Service'].value_counts()

F    6772
O       8
Name: Place of Service, dtype: int64

In [80]:
Filter7 = F6['Place of Service'].isin(['F'])

F7 = F6[Filter7]
F7.shape

(6772, 10)

Trying for a third time to merge with dataframe 2.  This time is works for me.

In [81]:
F7_with_d2= pd.merge(F7, d2, on=('APC'), how='inner')
F7_with_d2.shape

(8742652, 15)

In [82]:
F7_with_d2.head()

NPI Entity Type        Provider Type Place of Service HCPCS Code  \
0  1003006115           I  Physician Assistant                F      27130   
1  1003006115           I  Physician Assistant                F      27130   
2  1003006115           I  Physician Assistant                F      27130   
3  1003006115           I  Physician Assistant                F      27130   
4  1003006115           I  Physician Assistant                F      27130   

                                          HCPCS Desc  \
0  Replacement of thigh bone and hip joint prosth...   
1  Replacement of thigh bone and hip joint prosth...   
2  Replacement of thigh bone and hip joint prosth...   
3  Replacement of thigh bone and hip joint prosth...   
4  Replacement of thigh bone and hip joint prosth...   

   No. Distinct Medicare Beneficiaries Per Day Services  \
0                                                 14      
1                                                 14      
2                                                 14      
3                                                 14      
4                                                 14      

   Avg Medicare Amt Allowed                APC Desc     APC  Provider_ID  \
0                    230.51  Total hip arthroplasty  5115.0        10001   
1                    230.51  Total hip arthroplasty  5115.0        10005   
2                    230.51  Total hip arthroplasty  5115.0        10006   
3                    230.51  Total hip arthroplasty  5115.0        10011   
4                    230.51  Total hip arthroplasty  5115.0        10016   

                             APC_Desc  Beneficiaries  CAPC_Services  \
0  Level 5 Musculoskeletal Procedures           59.0             59   
1  Level 5 Musculoskeletal Procedures           25.0             25   
2  Level 5 Musculoskeletal Procedures           15.0             16   
3  Level 5 Musculoskeletal Procedures           21.0             22   
4  Level 5 Musculoskeletal Procedures           15.0             15   

   Average_Medicare_Allowed_Amount  
0                          8078.53  
1                          8321.96  
2                          7624.83  
3                          7993.85  
4                          7845.45

In [83]:
F7_with_d2.columns

Index(['NPI', 'Entity Type', 'Provider Type', 'Place of Service', 'HCPCS Code',
       'HCPCS Desc', 'No. Distinct Medicare Beneficiaries Per Day Services',
       'Avg Medicare Amt Allowed', 'APC Desc', 'APC', 'Provider_ID',
       'APC_Desc', 'Beneficiaries', 'CAPC_Services',
       'Average_Medicare_Allowed_Amount'],
      dtype='object')